### ISSUE TITLE
 role-has-required-aria-props warns about `role="switch"` missing `aria-checked` when used with native `<input checked>`

### ISSUE THREAD

**Issue by hoonweiting** [vue-a11y/eslint-plugin-vuejs-accessibility#932]
I have a custom switch built with `<input type="checkbox">`:

```vue
<input
  type="checkbox"
  role="switch"
  :checked="isChecked"
>
```

After enabling role-has-required-aria-props, the following problem is reported:

```
Elements with the ARIA role "switch" must have the following attributes defined: aria-checked
```

I believe this is a false positive. `<input type="checkbox">` already has its own `checked`, and I'd think we shouldn't need to replace it with `aria-checked`. A similar issue was reported for Svelte's ESLint plugin: https://github.com/sveltejs/svelte/issues/7837

Should we consider allowing `checked` in place of `aria-checked`?

**Comment by vhoyer** [comment]
ok, I think you are correct, we should allow `checked` as a replacement for "aria-checked" if the current element is an `input`, basing this decision on this link from wai-aria: https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/

PRs are welcomed, sorry for the late reply I was on vacation (and a bit burnt out, still am but regardless)
